internal/resources/webform/webform.js

```diff
@@ -2356,21 +2357,23 @@ window.initGRPCForm = function(services, invokeURI, metadataURI, debug) {
 
     $("#grpc-request-timeout input").focus(function() {
         var inp = this;
+        var origValue = $(inp).val();
         setValidation(inp, function() {
-            var num = + $(inp).val();
-            if (num === undefined) {
+            var val = $(inp).val();
+            if (val === "" || val === undefined) {
                 return;
             }
+            var num = Number(val);
             if (Number.isNaN(num)) {
-                $(inp).val(undefined);
+                $(inp).val(origValue);
```

Comment:
I guess one pitfall here is that the user could see the validation error message "numeric value required" while _also_ seeing the original numeric value in the input? The end result being "I see a thing telling me a numeric value is required, but I also see a numeric value in the input, wha?"

Am I reading this right?

Reply:
Interesting point. In practice, at least in Chrome, this isn't an issue. I think it won't actually re-render this DOM change until after the JS thread quits (and since `alert` is blocking/modal, it doesn't actually update the page until after the modal is dismissed). However, even if that were to change, I think a better fix is to include the invalid text in the error message. That way, even if they see it change back, it will show them the invalid string they had entered. Does that seem like a reasonable solution?

Reply:
Hmmm, actually, it looks like there is some intent that the bad value be left. It actually applies a CSS attribute, to highlight the input as invalid (using same code used to validate inputs in the request form and to highlight invalid fields). So I can change it to leave the value and then fix the CSS so it's properly highlights, and that will probably make for the best experience.